python/lsst/summit/utils/nightReport.py

```diff
@@ -0,0 +1,633 @@
+            for seqNum in sorted(self.data.keys()):
+                if self.data[seqNum]['observation_type'] not in offSkyObsTypes:
+                    return seqNum
+            return None
+
+        if method == 'heuristic':
+            # take the first cwfs image and return that
+            seqNums = self.getSeqNumsMatching(observation_type='cwfs')
+            return min(seqNums)
+
+    def printObsTable(self, **kwargs):
+        """Print a table of the days observations.
+
+        Parameters
+        ----------
+        **kwargs : `dict`
+            Filter the observation table according to seqNums which match these
+            {k: v} pairs. For example, to only print out science observations
+            pass ``observation_type='science'``.
+        """
+        seqNums = self.data.keys() if not kwargs else self.getSeqNumsMatching(**kwargs)
+        seqNums = sorted(seqNums)  # should always be sorted, but is a totaly disaster here if not
+
+        dts = self.getTimeDeltas()
+        lines = []
+        for seqNum in seqNums:
+            try:
+                expTime = self.data[seqNum]['exposure_time'].value
+                imageType = self.data[seqNum]['observation_type']
+                target = self.data[seqNum]['target_name']
+                deadtime = dts[seqNum]
+                filt = self.data[seqNum]['physical_filter']
+
+                msg = f'{seqNum} {target} {expTime:.1f} {deadtime:.02f} {imageType} {filt}'
+            except Exception:
+                msg = f"Error parsing {seqNum}!"
+            lines.append(msg)
+
+        print(r"seqNum target expTime deadtime imageType filt")
+        print(r"------ ------ ------- -------- --------- ----")
+        for line in lines:
+            print(line)
+
+    def getExposureMidpoint(self, seqNum):
+        """Return the midpoint of the exposure as a float in MJD.
+
+        Parameters
+        ----------
+        seqNum : `int`
+            The seqNum to get the midpoint for.
+
+        Returns
+        -------
+        midpointMjd : `float`
+            The midpoint, as an mjd float.
+        """
+        timespan = self.data[seqNum]['timespan']
+        return (timespan.begin.mjd + timespan.end.mjd) / 2
+
+    def plotPerObjectAirMass(self, objects=None, airmassOneAtTop=True, saveFig=''):
+        """Plot the airmass for objects observed over the course of the night.
+
+        Parameters
+        ----------
+        objects : `list` [`str`], optional
+            The objects to plot. If not provided, all objects are plotted.
+        airmassOneAtTop : `bool`, optional
+            Put the airmass of 1 at the top of the plot, like astronomers
+            expect.
+        saveFig : `str`, optional
+            Save the figure to this file path?
+        """
+        if not objects:
+            objects = self.stars
+
+        objects = ensure_iterable(objects)
+
+        _ = plt.figure(figsize=(10, 6))
```

Comment:
I'm assuming that at no point you are going to have two figures open so this doesn't matter?